### DISCUSSION TITLE
Class with methods

### DISCUSSION THREAD

**Discussion by igitur** [inducer/genpy#11] [Q&A]
Hi,

I'm checking this out for a possible code generation project. How would I generate a class with methods/functions?

thanks

**Comment by inducer** [comment]
You can try and use this test as a starting point:

https://github.com/inducer/genpy/blob/f03d18ab25b663d687e7393bec9c957e641667cb/test/test_genpy.py#L27-L38

**Reply by igitur** [reply]
Already saw that and I'm fine with Function. It was the missing colon for the Class that threw me off.

**Reply by igitur** [reply]
Oh, and I thought Class.attributes referred to decorators initially. Combined with the missing colon I couldn't figure out what was going on.
